ngx_http_proxy_connect_module.c

```diff
@@ -1712,10 +1712,14 @@ ngx_http_proxy_connect_set_address(ngx_http_request_t *r,
     }
 
     if (address->value == NULL) {
-        u->peer.sockaddr = address->addr->sockaddr;
-        u->peer.socklen = address->addr->socklen;
-        u->peer.name = &address->addr->name;
+
         u->resolved->naddrs = 1;
+        u->resolved->addrs = NULL;
+        u->resolved->sockaddr = address->addr->sockaddr;
+        u->resolved->socklen = address->addr->socklen;
+        u->resolved->name = address->addr->name;
+        u->resolved->host = address->addr->name;
```

Comment:
Note that we should not modified pc(ngx_event_peer_s) here, should modify it in pc->get.

Also note that this module's pc->get(ngx_http_proxy_connect_get_peer) is fake function, it initializes pc before pc->get is called.

```
ngx_http_proxy_connect_process_connect()
{
...
    pc->sockaddr = ur->sockaddr;
    pc->socklen = ur->socklen;
    pc->name = &ur->host;

    pc->get = ngx_http_proxy_connect_get_peer;

    rc = ngx_event_connect_peer(&u->peer);

```